packages/strapi-hook-mongoose/lib/index.js

```diff
@@ -336,8 +338,10 @@ module.exports = function (strapi) {
                     // Set this info to be able to see if this field is a real database's field.
                     details.isVirtual = true;
                   } else {
+                    const customKey = details.plugin ? strapi.plugins[details.plugin].models[details.model].attributes['_id'] : strapi.models[details.model].attributes["_id"];
```

Comment:
You should only do this once? I guess the logic to retrieve the customKey and convert to the right type should be placed before the switch statement.